src/main/java/uk/gov/pay/products/resources/PaymentResource.java

```diff
@@ -61,10 +64,22 @@ public Response findPaymentByExternalId(@PathParam("paymentExternalId") String p
     @Produces(APPLICATION_JSON)
     @Consumes(APPLICATION_JSON)
     @PermitAll
-    public Response createPaymentByProductExternalId(@PathParam("productExternalId") String productExternalId) {
+    public Response createPayment(@PathParam("productExternalId") String productExternalId, JsonNode priceOverride) {
         logger.info("Create a payment for product id - [ {} ]", productExternalId);
-        Payment payment = paymentFactory.paymentCreator().doCreate(productExternalId);
-        return Response.status(CREATED).entity(payment).build();
+        return requestValidator.validatePriceOverrideRequest(priceOverride)
+                .map(errors -> Response.status(Response.Status.BAD_REQUEST).entity(errors).build())
+                .orElseGet(() -> {
+                    Payment payment = paymentFactory.paymentCreator().doCreate(productExternalId, extractAmountIfAvailable(priceOverride));
+                    return Response.status(CREATED).entity(payment).build();
```

Comment:
What if Payment Creation fails?

Reply:
I believe it will be an 500. I didn't change the existing logic here. Only wrapped a validation around the existing logic